app/js/components/names.jsx

```diff
@@ -1,265 +1,251 @@
-            dateCreated: '',
-            res: '',
-        };
-        this.handleEdit = this.handleEdit.bind(this);
-        this.handleSave = this.handleSave.bind(this);
-        this.handleCancel = this.handleCancel.bind(this);
-        this.handleChange = this.handleChange.bind(this);
-        this.fetch = this.fetch.bind(this);
-    }
-
-    /**
-     *
-     * set to state patient attributes
-     *
-     * @memberOf Names
-     */
-    componentWillMount() {
-        this.fetch();
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: 'view',
+      givenName: '',
+      middleName: '',
+      familyName: '',
+      voided: false,
+      preferred: true,
+      creator: '',
+      dateCreated: '',
+      res: '',
+      error: ''
     };
-
-    /**
-     *
-     * make API call to get the values then set to state
-     *
-     * @memberOf Names
-     */
-    fetch(){
-        apiCall(null, 'get', 'patient/' + this.props.uuid + '?v=full').then((res) => {
-          this.setState({   res: res.person,
-              givenName: res.person.preferredName.givenName,
-              middleName: res.person.preferredName.middleName,
-              familyName: res.person.preferredName.familyName,
-              voided: res.person.preferredName.voided,
-              creator: res.person.auditInfo.creator.display,
-              dateCreated: res.person.auditInfo.dateCreated
-            });
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.fetch = this.fetch.bind(this);
+  }
+
+  componentWillMount() {
+    this.fetch();
+  };
+
+  fetch() {
+    apiCall(null, 'get', 'patient/' + this.props.uuid + '?v=full').then((res) => {
+      this.setState({
+        res: res.person,
+        givenName: res.person.preferredName.givenName,
+        middleName: res.person.preferredName.middleName,
+        familyName: res.person.preferredName.familyName,
+        voided: res.person.preferredName.voided,
+        creator: res.person.auditInfo.creator.display,
+        dateCreated: res.person.auditInfo.dateCreated,
+        errorGivenName: '',
+        errorFamilyName: ''
+      });
+    });
+  };
+
+  handleEdit() {
+    this.setState({ display: 'edit' });
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target
+    event.preventDefault();
+    this.setState({
+      [name]: value
+    });
+  }
+
+  handleSave() {
+    this.setState((prevState) => ({
+      givenName: this.refs.givenname.value,
+      middleName: this.refs.middlename.value,
+      familyName: this.refs.familyname.value,
+      creator: prevState.creator,
+      dateCreated: prevState.dateCreated,
+      errorGivenName: '',
+      errorFamilyName: ''
+    }));
+    if(this.state.givenName && this.state.familyName) {
+      apiCall({
+            'givenName': this.state.givenName,
```

Comment:
We could format here to use the correct indentation.

Reply:
Done